packages/gatsby/src/schema/types/__tests__/date.js

```diff
@@ -0,0 +1,453 @@
+const { store } = require(`../../../redux`)
+const { build } = require(`../..`)
+const { isDate } = require(`../date`)
+
+// Timestamps grabbed from https://github.com/moment/moment/blob/2e2a5b35439665d4b0200143d808a7c26d6cd30f/src/test/moment/is_valid.js
+
+describe(`isDate`, () => {
+  beforeEach(async () => {
+    // pass
+  })
+
+  it(`supports valid ISO 8601 datetimes`, async () => {
+    const tests = [
+      `1970`,
+      `2019`,
+      `1970-01`,
+      `2019-01`,
+      `1970-01-01`,
+      `2010-01-01`,
+      `2010-01-30`,
+      `19700101`,
+      `20100101`,
+      `20100130`,
+      `2010-01-30T23+00:00`,
+      `2010-01-30T23:59+00:00`,
+      `2010-01-30T23:59:59+00:00`,
+      `2010-01-30T23:59:59.999+00:00`,
+      `2010-01-30T23:59:59.999-07:00`,
+      `2010-01-30T00:00:00.000+07:00`,
+      `2010-01-30T23:59:59.999-07`,
+      `2010-01-30T00:00:00.000+07`,
+      `1970-01-01T00:00:00.000Z`,
+      `2012-04-01T00:00:00-05:00`,
+      `2012-11-12T00:00:00+01:00`,
+    ]
+    let i
+    for (i = 0; i < tests.length; i++) {
+      expect(isDate(tests[i])).toEqual(true)
+    }
+  })
+
+  it(`supports ISO datetimes without T`, async () => {
```

Comment:
none of these tests are actually using `await` or returning a `Promise`, so you shouldn't use an `async` function